core/variant_parser.cpp

```diff
@@ -744,7 +744,7 @@ Error VariantParser::parse_value(Token &token, Variant &value, Stream *p_stream,
 						return err;
 
 					if (token.type == TK_PARENTHESIS_CLOSE) {
-
+						value = REF(obj->cast_to<Reference>());
```

Comment:
Probably not.  I convinced myself that all Objects could be cast to a reference but looking back thats definitely false as Reference inherits from Object and not the other way around.  

Sorry I'm still new 😄 

Would something like this work?

```cpp
Reference *r = obj->cast_to<Reference>();
if (r)
{
    value = REF(r);
}
else
{
    value = obj;
}
```